aiidalab_widgets_base/computational_resources.py

```diff
@@ -1214,9 +1246,16 @@ def _observe_code_setup(self, _=None):
             if hasattr(self, key):
                 if key == "default_calc_job_plugin":
                     try:
-                        getattr(self, key).label = value
-                    except traitlets.TraitError:
-                        self.message = f"Input plugin {value} is not installed."
+                        self.default_calc_job_plugin.value = value
+                    except tl.TraitError:
+                        import re
```

Comment:
I am always not sure when to import the module, I guess import on the rare case what make the loading of the module a bit fast? Especially for the module that takes long time to load. 
I assume @danielhollas has more experience on this, after the effort of improve the loading time of literally everything :p  Is there a rule we can follow?

Reply:
Generally speaking, imports from stdlib are okay (with some exceptions). Also re is such a common module that it will for sure be imported from somewhere else anyway.

I'll try to write down more in a separate issue .